### ISSUE TITLE
Sentry.HTTPClient behaviour is too broad

### ISSUE THREAD

**Issue by josevalim** [getsentry/sentry-elixir#399]
I was looking for a way to replace Hackney by Finch and, while Sentry does provide a behaviour for it, `Sentry.HTTPClient` is too broad at the moment. I am thinking we could have a lower level API that handles only the HTTP parts. For example, in Swoosh, we need just something like:

```elixir
  def post(url, headers, body, %Swoosh.Email{}) do
    # Postmark adapter sends url as iodata.
    url = IO.iodata_to_binary(url)

    case Finch.request(Bytepack.Finch, :post, url, headers, body) do
      {:ok, %Finch.Response{} = response} ->
        {:ok, response.status, response.headers, response.body}

      {:error, reason} ->
        {:error, reason}
    end
  end
```

I would like to send a PR but I am not sure how to tackle backwards compatibility. Should we provide a new behaviour that will deprecate the current one? If so, what should it be called?

**Comment by mitchellhenke** [comment]
Yep, I think that would be a lot better. I want to do a major release soon, and so I think switching to a new default behaviour would be good to include in that. The current one can be deprecated as part of that, and eventually removed.

**Comment by mitchellhenke** [comment]
closed in #400 🙂
